crates/sui-rosetta/src/construction.rs

```diff
@@ -143,20 +143,27 @@ pub async fn submit(
     WithRejection(Json(request), _): WithRejection<Json<ConstructionSubmitRequest>, Error>,
 ) -> Result<TransactionIdentifierResponse, Error> {
     env.check_network_identifier(&request.network_identifier)?;
-    let signed_tx: Transaction = bcs::from_bytes(&request.signed_transaction.to_vec()?)?;
+    let signed_tx: sui_types::transaction::Transaction =
+        bcs::from_bytes(&request.signed_transaction.to_vec()?)?;
+
+    let signatures = signed_tx
+        .tx_signatures()
+        .iter()
+        .cloned()
+        .map(UserSignature::from)
+        .collect();
+
+    let proto_transaction: Transaction = signed_tx
+        .into_data()
+        .into_inner()
+        .intent_message
+        .value
+        .into();
```

Comment:
We only really need this converted to bcs if this is on the submission path, can we double check that this isn't populating all the other fields?